pkgs/development/tools/marksman/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, fetchFromGitHub
+, buildDotnetModule
+, dotnetCorePackages
+}:
+
+buildDotnetModule rec {
+  pname = "marksman";
+  version = "2022-09-13";
+
+  src = fetchFromGitHub {
+    owner = "artempyanykh";
+    repo = "marksman";
+    rev = version;
+    sha256 = "sha256-87XuapUeeIw+MQ8rosiG7p6SVNOmflBsb3CcvfaUdCU=";
+  };
+
+  patches = [ ./patches/0001-revert-embed-git-commit-into-the-version.patch ];
+
+  projectFile = "Marksman/Marksman.fsproj";
+
+  doCheck = true;
+  testProjectFile = "Tests/Tests.fsproj";
+
+  nugetDeps = ./deps.nix;
```

Comment:
Since this is a relatively complex update process, it would be really nice if there were an [update script](https://nixos.org/manual/nixpkgs/stable/#var-passthru-updateScript) for this one, especially if you don't intend to maintain it yourself.

I've never used dotnet either, but I imagine it wouldn't be too hard to do the nuget fetching in a script?